test/PlatformFunctions.spec.ts

```diff
@@ -181,5 +196,158 @@ for (const platform in platforms) {
         expect(await platformFunctions.getUname()).to.be.null;
       });
     });
+
+    describe('createEmitter', () => {
+      let emitter;
+      beforeEach(() => {
+        emitter = platformFunctions.createEmitter();
+      });
+
+      it('should emit a `foo` event with data to listeners', (done) => {
+        function onFoo(data): void {
+          emitter.removeListener('foo', onFoo);
+
+          expect(data.bar).to.equal('bar');
+          expect(data.baz).to.equal('baz');
+
+          done();
+        }
+
+        emitter.on('foo', onFoo);
+        emitter.emit('foo', {bar: 'bar', baz: 'baz'});
+      });
+
+      it('listeners registered via `on` keep firing until removed', (done) => {
+        let calls = 0;
+
+        function onFoo(): void {
+          calls += 1;
+        }
+
+        emitter.on('foo', onFoo);
+        emitter.emit('foo');
+        expect(calls).to.equal(1);
+        emitter.emit('foo');
+        expect(calls).to.equal(2);
+        emitter.removeListener('foo', onFoo);
+
+        done();
+      });
+
+      it('listeners registered via `once` only fire once', (done) => {
+        let calls = 0;
+
+        function onFoo(): void {
+          calls += 1;
+        }
+
+        emitter.once('foo', onFoo);
+        emitter.emit('foo');
+        expect(calls).to.equal(1);
+        emitter.emit('foo');
+        expect(calls).to.equal(1);
+
+        done();
+      });
+
+      it('listeners registered multiple times are for each time it was registered', (done) => {
+        let calls = 0;
+
+        function onFoo(): void {
+          calls += 1;
+        }
+
+        emitter.on('foo', onFoo);
+        emitter.once('foo', onFoo);
+
+        emitter.emit('foo');
+        expect(calls).to.equal(2);
+
+        done();
+      });
+
+      it('should not emit a `foo` event to removed listeners', (done) => {
+        function onFoo(): void {
+          done(new Error('How did you get here?'));
+        }
+
+        emitter.once('response', onFoo);
+        emitter.removeListener('response', onFoo);
+        emitter.emit('foo');
+
+        done();
+      });
+    });
+
+    describe('tryBufferData', () => {
+      if (!isNodeEnvironment) {
+        // WebPlatformFunctions
+        it('should throw an error in web environments if streaming data is provided', () => {
+          if (process.versions.node < '18') {
+            console.log(
+              `'ReadableStream' is not available in the global scope for ${process.version}, skipping test.`
+            );
+            return;
+          }
+
+          const f = new ReadableStream();
```

Comment:
This won't work on node as well. Should throw everywhere.

Reply:
This test doesn't pass in node, but actually I don't think it's worth checking / throwing if the user tries to pass in a ReadableStream. We haven't had any issues / user requests for it, and it would add an additional, unnecessary check in `tryBufferData`.

Reply:
I'll leave this up to you but we both know what the behavior is when passing a `ReadableStream` on node and it's bad.

Reply:
I agree it's bad since the `ReadableStream` gets stringified, but I think this puts too many guard rails in our library for something users are unlikely to do. Also, `ReadableStream` is not available in Node <18, so we'd have to use a duck-typing check to see if the file data looks like a `ReadableStream`. 🦆